### PR TITLE
Adding `Dry::Container::Mixin#each`

### PR DESCRIPTION
**Pull request by jeremyf**
In discussions with other developers, it was felt that being able to
iterate over not just the registered keys, but to see what was
registered would be very helpful. This is a step toward doing that.

### CODE REVIEW

File: lib/dry/container/mixin.rb [hunk 1]
```diff
@@ -183,6 +183,22 @@ def each_key(&block)
         self
       end
 
+      # Calls block once for each key/value pair in the container, passing the key and the registered item parameters.
+      #
+      # If no block is given, an enumerator is returned instead.
```
Comment (AMHOL) [inline lib/dry/container/mixin.rb:188]: This statement is not true with the current code as `#each` returns `self`, it should probably just return the result of `Resolver#each`

Comment (jeremyf) [reply lib/dry/container/mixin.rb:188]: @AMHOL Updated

**Comment by AMHOL** [comment]
Thanks @jeremyf

**Comment by solnic** [comment]
;)

**Comment by AMHOL** [comment]
@solnic done

**Pull request merged**